### PR TITLE
Local storage(Pair-programming)

### PR DESCRIPTION
**Pull request by Shabiblac**
### Milestine : **Local storage**

**In this branch, we:** 

- Added a local storage function.
- Added a function to store user details on the form.

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is not contributed by all (2) assigned pair group members. Please use realistic GitHub accounts and ensure that all assigned members to this project contribute evenly to this task, and then request another code review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by bobb-Rob — COMMENTED** [review]
Hi @Shabiblac, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted the wrong project pull request - **portfolio-localStorage**, instead of **Portfolio: details popup window**. KIndly re-submit the correct PR.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: script.js [hunk 1]
```diff
@@ -67,12 +67,34 @@ function validateEmail() {
...
+  formText.value = info.formText;
+};
 function validateForm() {
   if (!validate() || !validateEmail()) {
     submitError.style.display = 'block';
     submitError.innerHTML = 'Please fix errors to submit';
     setTimeout(() => { submitError.style.display = 'none'; }, 3000);
     return false;
-  } return true;
+  }
```
Comment (shella12) [inline script.js]: Great job on using the required data model however in this milestone you have to utilise local storage to persist data in your form and according to the [requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md#project-requirements), you should implement the following interactions:

- When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

- [ ] Currently no data is being saved to the local storage (You can check it using developer tools:  inspect   => application => localStorage). Please make sure when the user inputs something in the input field the data gets stored to the localStorage. And if the user reloads the page the data should remain in the input field. Example snippet:

```js
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi @Shabiblac and @Paulinelali, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Used the required data model ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by wayungi — APPROVED** [review]
Hi @Shabiblac , 

image

Great work, your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**Remember: Every finish line 🏁  is the beginning of a new race 🐎** 

## Highlights
- [ ] All requested changes implemented ✔️ 
- [ ] Form reloads data from local storage if previous filled ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**